nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -341,6 +341,44 @@ in
         ";
       };
 
+      logError = mkOption {
+        default = "stderr";
+        description = "
+          Configures logging.
+          The first parameter defines a file that will store the log. The
+          special value stderr selects the standard error file. Logging to
+          syslog can be configured by specifying the “syslog:” prefix.
+          The second parameter determines the level of logging, and can be
+          one of the following: debug, info, notice, warn, error, crit,
+          alert, or emerg. Log levels above are listed in the order of
+          increasing severity. Setting a certain log level will cause all
+          messages of the specified and more severe log levels to be logged.
+          If this parameter is omitted then error is used.
+        ";
+      };
+
+      daemon = mkOption {
+        default = false;
+        type = types.bool;
+        description = "
+          Determines whether nginx should become a daemon. Mainly used
+          during development.
+        ";
+      };
+
+      preStart =  mkOption {
+        type = types.lines;
+        default = ''
+          test -d ${cfg.stateDir}/logs || mkdir -m 750 -p ${cfg.stateDir}/logs
+          test `stat -c %a ${cfg.stateDir}` = "750" || chmod 750 ${cfg.stateDir}
+          test `stat -c %a ${cfg.stateDir}/logs` = "750" || chmod 750 ${cfg.stateDir}/logs
+          chown -R ${cfg.user}:${cfg.group} ${cfg.stateDir}
+        '';
+        description = "
+          Shell commands executed before the service's nginx is started.
+        ";
```

Comment:
What is the motivation for that? Can you provide an example?

Reply:
sorry, deleted messgae

Reply:
How does the `preStart` option help you here?

Reply:
There are no problems in this case. Normal worked.

Reply:
everything above the command 
`/nix/store/zg5sc4sby91mp4xvd29s9f1i7yc67va3-nginx-1.14.0/bin/nginx`
normal worked.